base/show.jl

```diff
@@ -266,7 +274,14 @@ function show_type_name(io::IO, tn::TypeName)
     elseif globfunc
         print(io, "typeof(")
     end
-    if isdefined(tn, :module) && !(is_exported_from_stdlib(sym, tn.module) || (tn.module === Main && !hidden))
+    # Print module prefix unless type is:
```

Comment:
Wasn't immediately clear to me without reading the code if this was a conjunction or disjunction of conditions, so maybe change the comment to this:
```jl
# Print module prefix unless type satisfies one of the following:
```
But see comment below which would change this to just check visibility.